Newest mypy causing build issues

mypy changes the isclass annotation causing the build issues

Build Error:
`
Running MyPy...
mypy --install-types --non-interactive
moto/core/models.py:68:41: error: Argument 1 to "_decorate_class" of "MockAWS" has incompatible type "type[object]"; expected "Callable[P, T]"  [arg-type]
Found 2 errors in 1 file (checked 1146 source files)
moto/core/models.py:192:16: error: Incompatible return value type (got "type[object]", expected "Callable[P, T]")  [return-value]
make: *** [Makefile:25: lint] Error 1
`

isclass annotation changed from `TypeIs[type[Any]]` to `TypeIs[type[object]]` in 2.3.0.

[mypy change](https://github.com/python/mypy/blame/82e8498c19494edebee08dd615470443ec53fffc/mypy/typeshed/stdlib/inspect.pyi#L220)

--- a/moto/core/models.py
+++ b/moto/core/models.py
@@ -12,6 +12,7 @@ from typing import (
     TYPE_CHECKING,
     Any,
     TypeVar,
+    cast,
 )
 from unittest.mock import patch
 
@@ -65,7 +66,7 @@ class MockAWS(AbstractContextManager["MockAWS"]):
         self, func: "Callable[P, T]", reset: bool = True, remove_data: bool = True
     ) -> "Callable[P, T]":
         if inspect.isclass(func):
-            return self._decorate_class(func)
+            return cast("Callable[P, T]", self._decorate_class(func))
         return self._decorate_callable(func, reset, remove_data)
 
     def __enter__(self) -> "MockAWS":
@@ -125,8 +126,7 @@ class MockAWS(AbstractContextManager["MockAWS"]):
         wrapper.__wrapped__ = func  # type: ignore[attr-defined]
         return wrapper
 
-    def _decorate_class(self, klass: "Callable[P, T]") -> "Callable[P, T]":
-        assert inspect.isclass(klass)  # Keep mypy happy
+    def _decorate_class(self, klass: type[T]) -> type[T]:
         direct_methods = get_direct_methods_of(klass)
         defined_classes = {x for x, y in klass.__dict__.items() if inspect.isclass(y)}
 

